sycl/include/sycl/ext/intel/esimd/detail/sycl_util.hpp

```diff
@@ -120,9 +120,10 @@ template <typename T>
 inline constexpr bool is_rw_accessor_v =
     is_rw_device_accessor_v<T> || is_rw_local_accessor_v<T>;
 
-template <typename T, accessor_mode_cap_val_t Capability, typename RetT>
-using EnableIfAccessor =
-    std::enable_if_t<detail::is_device_accessor_with_v<T, Capability>, RetT>;
+template <typename T, accessor_mode_cap_val_t Capability, typename RetT,
+          bool AdditionalParam = true>
```

Comment:
`ExtraCondition` probably would be a better name.

More general comment: From design point of view I'd rather oppose the idea of adding this param to `EnableIfAccessor`, it is a bit of abuse/overuse of the existing check for accessor.
Please see my other comment in simd_obj_impl.hpp - perhaps, using this param is not needed, if replace that check with more clear `is_device_accessor_with_v` there.